src/persistentstorage.cpp

```diff
@@ -258,3 +266,229 @@ void MysqlPersistentStorage::writelog(const std::string &id, const ObjectValue &
     }
 }
 #endif // HAVE_MYSQL
+
+
+#ifdef HAVE_INFLUXDB
+Logger& InfluxdbPersistentStorage::logger_m(Logger::getInstance("InfluxdbPersistentStorage"));
+
+bool _existsInArray(const Json::Value &json, const std::string &key)
+{
+    bool ret = false;
+    assert(json.isArray());
+    for (int i = 0; i != json.size(); i++)
+    {
+        if (json[i][0].asString() == key)
+            ret = true;
+    }
+    return ret;
+}
+
+bool InfluxdbPersistentStorage::createDB(const std::string &db, std::string &result)
+{
+    std::string query = "q=CREATE DATABASE ";
+    query.append(db);
+    return curlRequest(INFLUXDB_QUERY, db, query, result);
+}
+
+InfluxdbPersistentStorage::InfluxdbPersistentStorage(ticpp::Element* pConfig)
+{
+    uri_m = pConfig->GetAttribute("uri");
+    user_m = pConfig->GetAttribute("user");
+    pass_m = pConfig->GetAttribute("pass");
+    db_m = pConfig->GetAttribute("db");
+    persist_db_m = pConfig->GetAttribute("persist_db");
+
+    std::string resp;
+
+    if (curlRequest(INFLUXDB_QUERY, db_m, "q=SHOW DATABASES", resp))
+    {
+        Json::CharReaderBuilder builder;
+        Json::CharReader * reader = builder.newCharReader();
+        Json::Value root, val;
+        std::string errors;
+
+        if(!reader->parse(resp.c_str(), resp.c_str()+resp.size(), &root, &errors))
+        {
+            logger_m.errorStream() << "couldn't parse JSON! Errors: " << errors << endlog;
+        }
+        else
+        {
+            Json::Value databases;
+            resp.clear();
+            databases = root["results"][0]["series"][0]["values"];
+            if (!_existsInArray(databases, db_m))
+                if (!createDB(db_m, resp))
+                    logger_m.errorStream() << "couldn't create logging database: " << resp << endlog;
+            if (!_existsInArray(databases, persist_db_m))
+                if (!createDB(persist_db_m, resp))
+                    logger_m.errorStream() << "couldn't create persistence database: " << resp << endlog;
+        }
+    }
+}
+
+InfluxdbPersistentStorage::~InfluxdbPersistentStorage()
+{
+}
+
+static size_t _curlWriteCB(void *contents, size_t size, size_t nmemb, void *userp)
+{
+    ((std::string*)userp)->append((char*)contents, size * nmemb);
+    return size * nmemb;
+}
+
+#define INFLUXDB_EXTRA_DEBUG 0
+
+bool InfluxdbPersistentStorage::curlRequest(InfluxdbOperation_t oper, const std::string& db, const std::string& query, std::string& result)
+{
+    bool ret = false;
+    CURL *curl;
```

Comment:
@fraxinas I don't see any `#ifdef HAVE_LIBCURL` protecting this code. Are we sure the code builds without libcurl?
Before merging this PR, I'd like to add a CURL-less scenario to the CI build matrix but in the meantime, you may want to check yourself what happens without libcurl.

Reply:
Please check to see if my suggestion is allright:

https://github.com/fraxinas/linknx/blob/influxdb/src/linknx.cpp#L71

Reply:
@fraxinas I think we are almost there. But as I mentioned a while ago about the `--with-influxdb` you initially proposed, I don't think `HAVE_INFLUXDB` is the flag we want. If I properly understood your changes, InfluxDB is not a dependency. I mean, we do not link against some `libinfluxdb.so`. The actual dependencies are both JsonCpp and libcurl so that we can perform calls to the InfluxDB web API.
As a consequence, I would expect to read `#if HAVE_JSONCPP && HAVE_LIBCURL` everywhere you currently have `#ifdef HAVE_INFLUXDB`.
Note that I am not totally against defining a `SUPPORT_INFLUXDB` preprocessor symbol either (it is rather a matter of taste, I believe). So feel free to pursue this approach but in that case, you need to fix `configure.ac:253`: unless I am mistaken, `HAVE_INFLUXDB` is defined if jsoncpp is present even if libcurl is not.

Reply:
@fraxinas I modified the CI build matrix to include a job building linknx with jsoncpp but not libcurl. Doing so, we'll have better confidence that we can merge your changes.